### PR TITLE
API-15: Get a list of attributes 

### PR DESCRIPTION
**Pull request by momoss**
Get a list of attributes
https://akeneo.atlassian.net/browse/API-15

Call /api/rest/v1/attributes
Response 200

```
[{"code":"sku","type":"pim_catalog_identifier","group":"marketing","unique":true,"useable_as_grid_filter":true,"allowed_extensions":[],"metric_family":null,"default_metric_unit":null,"reference_data_name":null,"available_locales":[],"max_characters":null,"validation_rule":null,"validation_regexp":null,"wysiwyg_enabled":false,"number_min":null,"number_max":null,"decimals_allowed":false,"negative_allowed":false,"date_min":null,"date_max":null,"max_file_size":null,"minimum_input_length":0,"sort_order":0,"localizable":false,"scopable":false,"labels":{"de_DE":"SKU","en_US":"SKU","fr_FR":"SKU"}},{"code":"name","type":"pim_catalog_text","group":"marketing","unique":false,"useable_as_grid_filter":true,"allowed_extensions":[],"metric_family":null,"default_metric_unit":null,"reference_data_name":null,"available_locales":[],"max_characters":null,"validation_rule":null,"validation_regexp":null,"wysiwyg_enabled":false,"number_min":null,"number_max":null,"decimals_allowed":false,"negative_allowed":false,"date_min":null,"date_max":null,"max_file_size":null,"minimum_input_length":0,"sort_order":0,"localizable":false,"scopable":false,"labels":{"de_DE":"Name","en_US":"Name","fr_FR":"Nom"}}]
```

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | :negative_squared_cross_mark:
| Added Behats                      | :negative_squared_cross_mark:
| Added integration tests           | :white_check_mark:
| Changelog updated                 | :negative_squared_cross_mark:
| Review and 2 GTM                  | :clock1:
| Micro Demo to the PO (Story only) | :negative_squared_cross_mark:
| Migration script                  | :negative_squared_cross_mark:
| Tech Doc                          | :negative_squared_cross_mark:

:white_check_mark: Done and pass

:red_circle: Done but fail

:clock1: Done but pending

:negative_squared_cross_mark: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php [hunk 1]
```diff
@@ -0,0 +1,814 @@
...
+use Akeneo\Test\Integration\TestCase;
+use Symfony\Component\HttpFoundation\Response;
+
+class ListAttributeIntegration extends TestCase
+{
+    public function testListAttributes()
+    {
+        $client = static::createClient();
+
+        $client->request('GET', 'api/rest/v1/attributes?limit=50&page=1');
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php]: why get all elements ? A better option would be :
- to add a test without limit and page 1 specified in the url, to test that only 10 elements are returned.
- test with limit=5 and page=2 to test that you only get the attributes from 6 to 10 position.

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+                'minimum_input_length'   => 0,
+                'sort_order'             => 0,
+                'localizable'            => false,
+                'scopable'               => false,
+                'labels'                 => [],
+            ],
+        ];
+
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php]: assert that you have 10 items please

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+                'max_file_size'          => null,
+                'minimum_input_length'   => 0,
+                'sort_order'             => 0,
+                'localizable'            => false,
+                'scopable'               => false,
+                'labels'                 => [],
+            ],
+        ];
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php]: same here (assert 5 items only)

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php (hunk 2 above)
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php]: We will add some tests to count items when we will integrate pagination stuff

File: src/Pim/Bundle/ApiBundle/Resources/config/routing.yml [hunk 4]
```diff
@@ -8,4 +8,4 @@ family:
 
 attribute:
     resource: '@PimApiBundle/Resources/config/routing/attribute.yml'
-    prefix: /rest/v1/attributes
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/Resources/config/routing.yml]: wasn't it better like that?
seems more logical to me

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+
+class ListAttributeIntegration extends TestCase
+{
+    public function testListAttributes()
+    {
+        $client = static::createClient();
+
+        $client->request('GET', 'api/rest/v1/attributes');
+
+        $standardAttributes = [
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Rest/Attribute/ListAttributeIntegration.php]: beware with EE
I think we override the attribute to add a property
so maybe your test will fail on EE

maybe we should remove this override, don't know

File: src/Pim/Bundle/ApiBundle/Resources/config/routing.yml (hunk 4 above)
Comment (momoss) [inline src/Pim/Bundle/ApiBundle/Resources/config/routing.yml]: Explain IRL

**Pull request merged**